app/scripts/modules/core/cancelModal/cancelModal.controller.ts

```diff
@@ -0,0 +1,44 @@
+import {module} from 'angular';
+import {IModalServiceInstance} from 'angular-ui-bootstrap';
+import IScope = ng.IScope;
+
+require('./cancel.less');
+
+interface ICancelModalScope extends IScope {
+  state: any;
+  params: any;
+  errorMessage: string;
+}
+
+export class CancelModalCtrl {
+
+  constructor(public $scope: ICancelModalScope, private $uibModalInstance: IModalServiceInstance, private params: any) {
+    this.$scope.params = params;
+
+    this.$scope.state = {
+      submitting: false
+    };
+  }
+
+  formDisabled = () => this.$scope.state.submitting;
+
+  showError(exception: string): void {
+    this.$scope.state.error = true;
+    this.$scope.state.submitting = false;
+    this.$scope.errorMessage = exception;
+  }
+
+  confirm(): void {
+    if (!this.formDisabled()) {
```

Comment:
You mean like this: `if (!this.formDisabled() && !this.$scope.state.submitting) {`? I don't understand how that will make any difference, as the `formDisabled()` function just returns the value of `this.$scope.state.submitting`. You might argue that the whole `formDisabled` function could be replaced with submitting state though, both here and in the html...